src/search.ts

```diff
@@ -78,4 +78,9 @@ export interface Search {
      * See https://www.hl7.org/fhir/capabilitystatement.html
      */
     getCapabilities(): Promise<SearchCapabilityStatement>;
+    /**
+     * Verify that a search string use as criteria in Subscription resource is valid
+     * and does not use _revinclude, _include, _sort, _count and chained parameter
+     */
+    isSubscriptionSearchCriteriaValid(searchCriteria: string): Boolean;
```

Comment:
A boolean method may not be ideal here. If it is invalid you just get `false` and can't provide details of failure to the customer. 

Maybe change it to a `void` method that throws an exception for invalid inputs

Reply:
It may be ok to throw the `InvalidSearchParameterError` here instead of creating a new error type